modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/execution.scala

```diff
@@ -0,0 +1,71 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * This structure holds the current `Execution` under execution. It carries
+  * information about which `Action`s have been completed.
+  *
+  */
+case class Current(execution: Execution[Action \/ Result]) {
```

Comment:
I acknowledge it's fair to expect the left part to be some kind of error or failure. I was piggybacking on some functions the `\/` API provides but it's probably worth making this type a distinct one.